tango/workspace_server.py

```diff
@@ -1,20 +1,52 @@
 import logging
-from http.server import ThreadingHTTPServer, BaseHTTPRequestHandler
+from http.server import ThreadingHTTPServer, BaseHTTPRequestHandler, SimpleHTTPRequestHandler
 from threading import Thread
 from typing import Tuple
+from glob import glob
+import sys
+import json
 
 from tango.workspace import Workspace
 
-
 logger = logging.getLogger(__name__)
 
+# get json for the ui viz for a list of steps
+def run_map(run_name, workspace):
+    stepMap = workspace.registered_run(run_name)
+    steps = {stepMap.get(sub).unique_id : serialize_step(stepMap.get(sub)) for sub in list(stepMap)}
+    return steps
 
-class WorkspaceRequestHandler(BaseHTTPRequestHandler):
-    def do_GET(self):
-        self.send_response(200)
-        self.end_headers()
-        self.wfile.write("Let's Tango!".encode("UTF-8"))
+def serialize_step(step):
+    return {
+        "unique_id": step.unique_id,
+        "step_name": step.step_name,
+        "step_class_name": step.step_class_name,
+        "version": step.version,
+        "dependencies": list(step.dependencies),
+        "start_time": step.start_time.isoformat(),
+        "end_time": step.end_time.isoformat(),
+        "error": step.error,
+        "result_location": step.result_location,
+        "status": step.status
+    }
 
+class WorkspaceRequestHandler(SimpleHTTPRequestHandler):
+    def do_GET(self):
+        print(self.path)
+        referer = self.headers.get('Referer')
+        if referer is None:
+            self.path = 'report/index.html'
```

Comment:
So if I request `/something.txt`, and there is no referrer, then you'll show the contents of the file `report/index.html`.

But if I follow a link to `/something`, then you'll show the contents of `report/something.txt`?

Sorry, I'm not clear on the intention here! I'm probably missing something obvious.

Reply:
ill just convert it tot '/'
thats the intent

Reply:
done